src/oneD/Flow1D.cpp

```diff
@@ -827,37 +842,48 @@ string Flow1D::componentName(size_t n) const
     }
 }
 
-size_t Flow1D::componentIndex(const string& name) const
+size_t Flow1D::componentIndex(const string& name, bool checkAlias) const
 {
-    if (name=="velocity") {
-        return c_offset_U;
-    } else if (name=="spread_rate") {
-        return c_offset_V;
-    } else if (name=="T") {
-        return c_offset_T;
-    } else if (name=="lambda") {
-        return c_offset_L;
-    } else if (name == "eField") {
-        return c_offset_E;
-    } else if (name == "Uo") {
-        return c_offset_Uo;
-    } else {
-        for (size_t n=c_offset_Y; n<m_nsp+c_offset_Y; n++) {
-            if (componentName(n)==name) {
-                return n;
-            }
+    if (componentMap.count(name)) {
+        return componentMap.at(name);
+    }
+    for (size_t n=c_offset_Y; n<m_nsp+c_offset_Y; n++) {
+        if (componentName(n)==name) {
+            return n;
+        }
+    }
+    if (checkAlias) {
+        auto aliasMap = _componentAliasMap();
```

Comment:
To avoid making a copy:
```suggestion
        const auto& aliasMap = _componentAliasMap();
```